src/standalone/vector.rs

```diff
@@ -215,6 +256,37 @@ impl VectorOps for StandaloneVectorClient {
         };
         Ok(ironcore_documents::key_id_header::get_prefix_bytes_for_search(key_id_header).into())
     }
+
+    async fn rotate_vectors(
+        &self,
+        encrypted_vectors: EncryptedVectors,
+        metadata: &AlloyMetadata,
+        new_tenant_id: Option<TenantId>,
+    ) -> RotateResult {
+        let new_metadata = match &new_tenant_id {
+            None => metadata.clone(),
+            Some(tenant_id) => AlloyMetadata {
+                tenant_id: tenant_id.clone(),
+                ..metadata.clone()
+            },
+        };
+        let attempts: Vec<_> = join_all(encrypted_vectors.into_iter().map(
+            |(vector_id, encrypted_vector)| {
+                self.rotate_vector(encrypted_vector, metadata, &new_metadata)
+                    .map(|rotated_vector| (vector_id, rotated_vector))
+            },
+        ))
+        .await;
+        let (rotate_successes, rotate_failures): (Vec<_>, Vec<_>) =
+            attempts.into_iter().partition_map(|r| match r {
+                (vector_id, Ok(rotated_vector)) => Either::Left((vector_id, rotated_vector)),
+                (vector_id, Err(e)) => Either::Right((vector_id, e.to_string())),
+            });
+        RotateResult {
+            successes: rotate_successes.into_iter().collect(),
+            failures: rotate_failures.into_iter().collect(),
+        }
+    }
```

Comment:
If we change `crate::util::hash_map_to_batch_result` to take `I: IntoIterator<Item = (FieldId, T)>` instead of locked to HashMap, we could use it with `std::convert::identity` to remove this duplication.

Just a suggestion though, I'm fine with you leaving it how it is